### PR TITLE
enhance: refactor leader_observer to leader_checker

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #29453 

sync distribution by rpc will also call loadSegment/releaseSegment, which may cause all kinds of concurrent case on same segment, such as concurrent load and release on one segment.
This PR add leader_checker which generate load/release task to correct the leader view, instead of calling sync distribution by rpc

### CODE REVIEW

File: internal/querycoordv2/checkers/leader_checker.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+		leaderWithOldVersion := version.GetVersion() < s.Version
+		// leader has newer version, but the query node which loaded the newer version has been shutdown
+		leaderWithDirtyVersion := version.GetVersion() > s.Version && c.nodeMgr.Get(version.GetNodeID()) == nil
+
+		if !ok || leaderWithOldVersion || leaderWithDirtyVersion {
+			log.Debug("leader checker append a segment to set",
+				zap.Int64("segmentID", s.GetID()),
+				zap.Int64("nodeID", s.Node))
+			action := task.NewSegmentActionWithScope(s.Node, task.ActionTypeGrow, s.GetInsertChannel(), s.GetID(), querypb.DataScope_Historical)
```
Comment (yah01) [inline internal/querycoordv2/checkers/leader_checker.go:132]: Suggest to add a new action type as:
- This impl changes the load scope to `Delta` case by case
- QueryNode doesn't impl `remove segment from leader view` as `ReleaseSegment`, but this PR does changed to

Comment (weiliu1031) [reply internal/querycoordv2/checkers/leader_checker.go:132]: 1. we need to set load scope for now case by case, even a new action type exist. such as the `SegmentUpdateAction`
2. actually query node does remove segment from leader view by `ReleaseSegment`

**Pull request merged**